src/System.Memory/src/System/Buffers/ArrayMemoryPool.ArrayMemoryPoolBuffer.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.Buffers
+{
+    internal sealed partial class ArrayMemoryPool<T> : MemoryPool<T>
+    {
+        private sealed class ArrayMemoryPoolBuffer : OwnedMemory<T>
+        {
+            private T[] _array;
+            private bool _disposed;
+            private int _referenceCount;
+
+            public ArrayMemoryPoolBuffer(int size)
+            {
+                _array = ArrayPool<T>.Shared.Rent(size);
+            }
+
+            public sealed override int Length => _array.Length;
```

Comment:
Do properties and methods on a sealed class need to marked as sealed as well?

Reply:
> Do properties and methods on a sealed class need to marked as sealed as well?

No